### DISCUSSION TITLE
Add Application blacklist 🚀

### DISCUSSION THREAD

**Discussion by lorre0322** [davatorium/rofi#1760] [General]
### Before opening a feature request

- [X] I checked the *next* branch to see if the feature has already been implemented
- [X] I searched existing reports to see if it is already requested.

### What is the user problem or growth opportunity you want to see solved?

I hope to add a blacklist of applications and make it not displayed in rofi through the path.

### How do you know that this problem exists today? Why is this important?

This is my first time to use rofi. It shows many applications I don't need, such as electron.
So I request to add application blacklist settings in the future

### Who will benefit from it?

everybody~!!💓

### Rofi version (rofi -v)

Version:1.7.5

### Configuration

rofi -dump-config

**Comment by DaveDavenport** [comment]
This can be done 'correctly' by copying the desktop file you want to hide into your user application directory and set the nodisplay property to true.
Rofi will parse this first, and hide the entry.

This will then work for every launcher that implements the standard.

The archlinux wiki is a good reference on this: https://wiki.archlinux.org/title/Desktop_entries

There used to be a graphical tool todo this, but not sure how it is called. (maybe somebody can make a rofi plugin for this?)

**Comment by petervaro** [comment]
I don't believe that is the _correct_ way of doing this, @DaveDavenport, because that requires duplicating the desktop-entries.  These files are regularly[^1] updated by the package maintainers for various reasons, which means that one either has to generate these files in an "update-hook" by extending them with the required property&mdash;which is quite a lot of faff to get done reliably&mdash;or (semi-)manually monitoring and updating them when necessary&mdash;which might work for a short while and list, but does not scale well beyond these&mdash;just so that they are kept in sync.

I think these are needlessly complicated alternatives for those who just want to configure their operating systems once and not constantly fight with them.  Especially so, when expressing any kind of listing in the configuration is supposed to be quite a simple task to support, and consolidating this list with other configurations&mdash;such as what you proposed&mdash;is trivial once the order of precedence has been established and documented.

Or am I not aware of a simple trick which would render the above complications void?

[^1]: Of course, not as frequently as the other bits of a package, still, I had my fair share of troubles with such things in the past, even if that is just an _anecdotal_ evidence.

**Reply by DaveDavenport** [reply]
the file does not need to be kept up2date, it just needs to resolve to the same desktop id.
https://whot.pages.freedesktop.org/xdg-specs/desktop-entry-spec/latest/ar01s02.html#desktop-file-id

It is the correct way to do it, and it should work independent of application launcher used.


If somebody wants to provide a script that uses rofi to hide/unhide desktop files I'll happily ship it with rofi.

**Reply by DaveDavenport** [reply]
On the previously linked arch wiki page it shows a list of menu editors that can do this for you.
This one seems to work well: https://github.com/donadigo/appeditor

**Reply by petervaro** [reply]
> [T]he file does not need to be kept up2date, it just needs to resolve to the same desktop id.

Well, that is not entirely true.  Indeed I was very glad you brought this up, I thought, 'Alright, all I need is the correct filename, one section and a single property and Bob's your uncle,' however, as it turned out the desktop-entry still needs to be valid, so one needs to have at least the `Type`, `Name`, and `Exec` properties too in order to make this work.

Still, I stand corrected, I didn't see the wood for the trees, indeed the desktop entries can go out of sync as long as their IDs match and we'd never ever wish to use those programs which we've _overridden_ the desktop-entries of for anything other than hiding them.  As such, this might sound like the _correct_ way of doing this.

(That being said&mdash;_old man grumbles_&mdash;it just does not sit well with me, that instead of having a single file where I can just list all the IDs I wish to show or hide, I have to maintain as many as I need to be filtered.)

**Reply by DaveDavenport** [reply]
That statement is not entirely true, as I never said the file did not have to be valid!
As mentioned in the initial post from me I said:
> This can be done 'correctly' by copying the desktop file you want to hide into your user application directory and set the nodisplay property to true.

Then I said that you don't need to update it on any change, as long as  resolves to the same desktop id.  I am unsure where I suggested you to make a new non-valid file?



I don't want to add functionality and maintain functionality to rofi for something that already has a (generic) solution and relevant tools.  The existing solution is also flexible, as you can specify you want hidden in environment X but not in Y.. (OnlyShowIn, NotShowIn).
